pubtools/_pulp/tasks/fix_cves.py

```diff
@@ -0,0 +1,171 @@
+from pubtools._pulp.arguments import SplitAndExtend
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+NOCHANNEL_REPO_MAP = {
+    "errata": "all-rpm-content",
+    "rpm": "all-rpm-content",
+    "iso": "all-iso-content",
+    "docker": "redhat-everything",
+    "docker_signature": "redhat-sigstore",
+}
+
+
+class FixCves(PulpTask, PulpClientService, Publisher):
+    """Command to fix cves"""
+
+    def fail(self, *args, **kwargs):
+        LOG.error(*args, **kwargs)
+        sys.exit(30)
+
+    @step("Get erratum")
+    def get_erratum(self, advisory_id):
+        erratum_page = self.get_erratum_from_pulp(advisory_id)
+        erratum = [ert for ert in erratum_page]
+        if not erratum:
+            self.fail("No erratum found for %s", advisory_id)
+
+        return erratum[0]
+
+    def get_erratum_from_pulp(self, advisory_id):
+        crit = Criteria.and_(
+            Criteria.with_unit_type(ErratumUnit), Criteria.with_id(advisory_id)
+        )
+        return self.pulp_client.search_content(criteria=crit).result()
+
+    @step("Process erratum")
+    def process_erratum_for_upload(self, erratum, cves):
+        # get non-cve erratum references
+        refs = [ref for ref in erratum.references if ref.type != "cve"]
+
+        # add new cves to erratum
+        for cve in cves:
+            refs.append(
+                ErratumReference(
+                    type="cve",
+                    title=cve,
+                    id=cve,
+                    href="https://www.redhat.com/security/data/cve/%s.html" % cve,
+                )
+            )
+
+        # updated version
+        version = int(erratum.version or "1") + 1
+
+        # update erratum
+        # clear repository memberships. It will be evaluated by Pulp.
+        erratum = attr.evolve(
+            erratum, references=refs, repository_memberships=None, version=str(version)
+        )
+
+        LOG.debug("Processed erratum: %s", erratum._to_data())
+
+        return erratum
+
+    @step("Upload erratum")
+    def upload_erratum(self, erratum):
+        # find no channel repo
+        repo_id = self.get_nochannel_repo_id("errata")
+
+        LOG.info("Getting nochannel repo %s", repo_id)
+        # get repo
+        repo = self.get_repo_from_pulp(repo_id)[0]
+
+        # upload to nochannel repo
+        repo.upload_erratum(erratum).result()
+        LOG.info("Uploaded to %s", repo_id)
+
+    def get_repo_from_pulp(self, repo_id):
+        crit = Criteria.with_id(repo_id)
+        resp = self.pulp_client.search_repository(criteria=crit)
+
+        return [r for r in resp.result()]
+
+    def get_nochannel_repo_id(self, repo_type):
+        return NOCHANNEL_REPO_MAP[repo_type]
+
+    @step("Get affected repos")
+    def get_affected_repos(self, erratum):
+        # get repos listed in repository memberships
+        repo_ids = erratum.repository_memberships
+        repo_ids = set(repo_ids) - set(NOCHANNEL_REPO_MAP.values())
```

Comment:
Removing this return the nochannel repos too in the list of affected repos. That list is later used to pick one of the random repos to upload which may return the nochannel repo as the destination to upload. However, it will definitely reduce the serialization.